tests/test_nsjail.py

```diff
@@ -174,3 +174,12 @@ def test_output_order(self):
             msg="stdout does not come before stderr"
         )
         self.assertEqual(result.stderr, None)
+
+    def test_stdout_flood_results_in_graceful_sigterm(self):
+        stdout_flood = dedent("""
+            while True:
+                print('abcdefghij')
+        """).strip()
+
+        result = self.nsjail.python3(stdout_flood)
+        self.assertEqual(result.returncode, -9)
```

Comment:
Is -9 equivalent to 137? Is this some sort of two's complement thing around 127 or 1 byte?

Reply:
No, it's what you get back when you terminate a subprocess with a signal in Popen:  [the signal code, but negative](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.returncode). So, that's what we're getting here as well, a `-9`, because SIGKILL is `9`.

Reply:
Then this needs to be adjusted somewhere so that the bot will understand that it's still `SIGKILL`, because right now I believe it will only understand `137`.

Reply:
We could return a manual return code from the consume function:
- If we exceed the limit, we return a manual 137
- If we don't, we propagate the actual exit code.

Reply:
How about something more generic to normalise it:

```py
if result.returncode < 0:
    result.returncode = abs(result.returncode) + 128
```

Reply:
I guess `-` can be used instead of `abs()`...